zookeeper/source/zk_dn_client.cc

```diff
@@ -200,15 +208,118 @@ namespace zkclient{
 		}
 
 		// Register the replication watcher for this dn
-		std::vector <std::string> children = std::vector <std::string>();
-		if(!zk->wget_children(queueName + id, children, watchFuncPtr, nullptr, error_code)){
+		std::vector <std::string> children;
+
+		if(!zk->wget_children(queueName + id, children, watchFuncPtr, this, error_code)){
 			LOG(INFO) << "getting children failed";
 		}
 
+		// While we still have work items continue to pull them off
+		while (children.size() > 0) {
```

Comment:
Does this need to be in a while loop? Since we already reattached the watcher. @Swyrough @Vegetable26